storage/objects/stream_file_upload.go

```diff
@@ -43,8 +43,21 @@ func streamFileUpload(w io.Writer, bucket, object string) error {
 	ctx, cancel := context.WithTimeout(ctx, time.Second*50)
 	defer cancel()
 
+	o := client.Bucket(bucket).Object(object)
+
+	// Add preconditions so that the upload may safely retry on transient errors.
```

Comment:
After some thought, I think we should add a link to the Go ref docs here so the user can understand what "generation" means. So:

"Add preconditions so that the upload may safely retry on transient errors. See pkg.go.dev/cloud.google.com/go/storage#hdr-Conditions for information on preconditions."

wdyt?

Reply:
We're going to need to include this comment in every sample with a precondition, and there are a lot of them. Would it make more sense to include these links in one place (probably a new section of the generations & preconditions docs) rather than putting them in a comment in each sample?